### ISSUE TITLE
GDALRasterBand::GetMetadataItem is too noisy since 3.7.0

### ISSUE THREAD

**Issue by sgillies** [OSGeo/gdal#7754]
The test at https://github.com/OSGeo/gdal/blob/1e23a32efcba827e591d6a198bbb2af609a22ca8/gcore/gdalrasterband.cpp#L8298-L8300 is too loose and causes needless warnings from uint8 type files which do not use `SIGNEDBYTE`.

**Comment by sgillies** [comment]
@rouault oh, the entire PIXELTYPE metadata item is deprecated? Now I get it! Sorry for the noise and useless complaining.

**Comment by rouault** [comment]
> oh, the entire PIXELTYPE metadata item is deprecated?

yes, in practice the only value possible for PIXELTYPE was SIGNEDBYTE
